lib/index.js

```diff
@@ -9,54 +9,114 @@
 var st = {};
 module.exports = st;
 
+/****
+ * Seems to be lots of different ways to format FASTA headers. 
+ * 
+ * Generally there's an ID and a DESCRIPTION
+ *   >ID DESCRIPTION
+ * 
+ *   >(parts|of|ID) (DESCRIPTION with optional key=values)
+ *   
+ * This is complicated by the fact that the "values" in the description can have spaces
+ * e.g. OS=Arabidopsis thaliana GN=CCD8
+ * 
+ ****
+*/
+
 // extract IDs and push them to the meta dict
 st.getMeta = function(label) {
-  if (st.contains(label, "|")) {
-    var identifiers = label.split("|");
-    var k = 0;
-    var database, databaseID;
-    var meta = {};
-    while (k < identifiers.length - 1) {
-      database = identifiers[k];
-      databaseID = identifiers[k + 1];
-      meta[database] = databaseID;
-      k += 2;
-    }
-    // assume the last entry is the label
-    var name = identifiers[identifiers.length - 1];
-    // check whether there is a uniprot id
-    if (name.indexOf("=") >= 0 && name.indexOf("OS") >= 0) {
-      var ds = {};
-      var details = name.split(" ");
-      ds.en = details[0];
-      details = details.splice(1);
-      var nameLength = findSepInArr(details, "=");
-      var detailsTmp = details.splice(nameLength - 1);
-      name = details.join(" ");
-      details = detailsTmp;
-      k = 0;
-      var block = [];
-      details.forEach(function(item) {
-        block.push(item);
-        if (item.indexOf("=") >= 0) {
-          strToDict(block.join(" "), "=", ds);
-          block = [];
-        }
-      });
-      return {
-        name: name,
-        ids: meta,
-        details: ds
-      };
-    }
-    return {
-      name: name,
-      ids: meta
-    };
-  }
-  return {
-    name: label
-  };
+  
+	var full_id = false, full_desc = false;
+	var name, ids = {}, details = {}, description;
+
+// 	console.log( "getMeta.label: ", label );
```

Comment:
Don't mind either way - I left them there because I sometimes find myself revisiting code and end up adding exactly the same debug comments that I deleted prior to the last commit :)